presto-memsql/src/main/java/io/prestosql/plugin/memsql/MemSqlClient.java

```diff
@@ -88,6 +132,118 @@ protected String getTableSchemaName(ResultSet resultSet)
+            // TODO handle decimals if decimal_rounding_mode session property is needed
+        }
+
+        // TODO add explicit mappings
+        return super.toPrestoType(session, connection, typeHandle);
+    }
+
+    @Override
+    public WriteMapping toWriteMapping(ConnectorSession session, Type type)
+    {
+        if (REAL.equals(type)) {
+            return WriteMapping.longMapping("float", realWriteFunction());
+        }
+        // TODO implement TIME types
+        // TODO add support for other TIMESTAMP precisions
+        if (TIMESTAMP_MILLIS.equals(type)) {
+            return WriteMapping.longMapping("datetime", timestampWriteFunction(TIMESTAMP_MILLIS));
+        }
+        if (VARBINARY.equals(type)) {
+            return WriteMapping.sliceMapping("mediumblob", varbinaryWriteFunction());
+        }
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            String dataType;
+            if (varcharType.isUnbounded()) {
+                dataType = "longtext";
+            }
+            else if (varcharType.getBoundedLength() <= 255) {
+                dataType = "tinytext";
+            }
+            else if (varcharType.getBoundedLength() <= 65535) {
+                dataType = "text";
+            }
+            else if (varcharType.getBoundedLength() <= 16777215) {
+                dataType = "mediumtext";
+            }
+            else {
+                dataType = "longtext";
+            }
+            return WriteMapping.sliceMapping(dataType, varcharWriteFunction());
+        }
+        if (type.equals(jsonType)) {
+            return WriteMapping.sliceMapping("json", varcharWriteFunction());
+        }
+
+        return super.toWriteMapping(session, type);
+    }
+
+    @Override
+    public void renameColumn(JdbcIdentity identity, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String newColumnName)
+    {
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            DatabaseMetaData metadata = connection.getMetaData();
+            if (metadata.storesUpperCaseIdentifiers()) {
+                newColumnName = newColumnName.toUpperCase(ENGLISH);
+            }
+            String sql = format(
+                    "ALTER TABLE %s CHANGE %s %s",
+                    quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTableName()),
+                    quoted(jdbcColumn.getColumnName()),
+                    quoted(newColumnName));
+            execute(connection, sql);
+        }
+        catch (SQLException e) {
+            // MemSQL versions earlier than 5.7 do not support the above CHANGE syntax
+            // TODO Does importing MySQL driver just for SQL_STATE_SYNTAX_ERROR make sense?
+            if ("42000".equals(e.getSQLState())) {
+                throw new PrestoException(NOT_SUPPORTED, format("Rename column not supported in catalog: '%s'", handle.getCatalogName()), e);
```

Comment:
this is nice (not supported vs error), but may be an overkill, as memsql 5.7 is hopefully passe.

i'd move `MemSQL versions earlier than 5.7 do not support the above CHANGE syntax` as a comment above where the syntax is used and remove the `if`  here.